src/sentry/models/organizationmemberteamreplica.py

```diff
@@ -0,0 +1,37 @@
+from django.db import models
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import (
+    BaseModel,
+    BoundedAutoField,
+    BoundedBigIntegerField,
+    control_silo_only_model,
+    sane_repr,
+)
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+
+
+@control_silo_only_model
+class OrganizationMemberTeamReplica(BaseModel):
+    """
+    Identifies relationships between organization members and the teams they are on.
+    """
+
+    __relocation_scope__ = RelocationScope.Excluded
```

Comment:
Three options:
1.  Replicas simply rerun all outbox replication logics.  This requires no work other than ensuring their redis cache version is reset to 0.
2.  Backup allow the post hook to create corresponding outboxes
3.  Backup create bulk outboxes using new bulk methods anyways

Idealy, there is no value in capturing replicated models because at any point, replicated models are only partially replicated (inbound outboxes likely exist).  Capturing the full closure of dependency is really easy to get wrong, *but relying on the intrinsic property that these objects are always a deterministic process that can be rerun seems more safe*.

Reply:
Makes sense, 1 sounds like the ideal solution to me - will that happen automatically right now when importing and organization or is that something we need to handle?

Reply:
@azaslavsky might help us answer that because I'm not sure lol.

Basically, we have some models (like OrganizationMemberMapping, TeamReplica, etc) that are created as a side effect of a post_save from other models.  We have several valid ways of handling them in the back up I mentioned above.  We're curious what your thoughts on desired goal should be.

Ideally we don't backup replicated models, we allow them to be "volumized" again via the outbox system.  If post_save is a problem, that's also fine so long as we can rely on the existing `backfill_outbox` job that would create async work to backfill them anyways.

It highly depends on what you seeing the goal of the backup and restore operations being.  If we rely on the async bbackfill job to recreate these objects, it means that restore won't truly function until "some async job" has succeeded.  If we want the backup job to guarantee the system is restored, we need to create and flush these outboxes (either through the post_save or via bulk operations, or somethihng else).

Reply:
Sorry about the late reply (fighting a fire elsewhere).

Am I correct to understand that this is a new Control Silo "shadow" model, serving the same function for `OrganizationMemberTeam` as `OrganizationMapping` does for `Organization` and `OrganizationMemberMapping` does for `OrganizationMember`?

If so, the way both `Organization` and `OrganizationMember` work is using the exact `post_save()` mechanism you describe: the models are excluded from the backup, and "rehydrated" when their regional source is imported. There are several reasons for this:

1. Though we don't really faithfully implement this idea today, the basic idea of a restore is "let's speed run all of the API calls that put the models there in the first place". This allows us to re-use as much logic (validation, post_save, etc) as possible when importing stuff. The fewer new codepaths we make for this feature, the better.

2. It's better to have one source of truth, rather than 2. For cases like control/region siloing, some data duplication is unavoidable. But the entire export bottoms out in a single JSON file, so it's unnecessary to save the information twice there.

3. Imports will never be atomic anyway, both due to the upcoming "siloing", and because once we include things like events we'll probably have too much data to make it practical. At that point, we'll need some way to manage imports in a step-wise manner regardless, so the ideal of "do this one weird command and get your database flashed into some known good state" won't really be possible. Importing this much data just kind of has to be async, so we might as well leverage the well-tested and commonly used codepaths we already where it makes sense to do so.

The main way we're solving potential concurrency issues is to have some key "root" models like `User` and `Organization` carry a new `is_unclaimed` flag, which is basically a signal that the model is in some way unstable post-import. So in the `Organization` case, we'd probably check to make sure the requisite `OrganizationMapping` exists before allowing  the model to be "claimed". We could consider doing the same here if we feel like the `post_save` asyncness is becoming a problem.